param/__init__.py

```diff
@@ -1201,6 +1203,178 @@ def get_range(self):
         raise NotImplementedError("get_range() must be implemented in subclasses.")
 
 
+
+class SelectObjects(list):
+    """
+    Wrapper around Selector.objects which allows both list-like and
+    dictionary like updates to the set of objects, ensuring backward
+    compatibility for setting objects as lists and dictionaries.
```

Comment:
I think a class should be documented by what it does and what it assumes, rather than by how it is used elsewhere in the code. That way this class can be defended and understood on its own, and potentially reused in other cases, rather than solely being tied to the specific way it's used in Selector.

```suggestion
    Container that supports both list-style and dictionary-style updates. Useful for replacing code that originally accepted lists but needs to support dictionary access (typically for naming items).
```